core-common/src/main/java/org/zalando/nakadi/service/subscription/zk/ZkSubscriptionClient.java

```diff
@@ -202,21 +203,18 @@ public Partition[] getPartitions() {
         }
 
         public Topology withUpdatedPartitions(final Partition[] partitions) {
-            final Partition[] resultPartitions = Arrays.copyOf(this.partitions, this.partitions.length);
+            final var resultPartitions = new ArrayList<>(Arrays.asList(this.partitions));
             for (final Partition newValue : partitions) {
```

Comment:
Since we are changing the implementation here, I'd rather go with making two maps and doing something like: `oldPartitions.addAll(newPartitions)`.